Add basic files for Pizza Order exercise (#2160)
exercism/javascript: exercises/concept/pizza-order/.docs/introduction.md

# Introduction

> The ability for something to be defined in terms of itself is called recursion. Recursive functions are functions that call themselves.

Reviewer: ```suggestion
The ability for something to be defined in terms of itself is called recursion.
Recursive functions are functions that call themselves.
```

> Suppose that you have a function called `recurse`. This function is recursive if it calls itself inside its body, like this:

Reviewer: ```suggestion
Suppose that you have a function called `recurse`.
This function is recursive if it calls itself inside its body, like this:
```

```js
function recurse() {
  // ...
  recurse();
  // ...
}
```

> A recursive function should always have a condition to stop calling itself and return a value, known as a _base case_. Otherwise, it will call itself indefinitely. In the absence of a base case, or when a function calls itself too many times (the number varies per the compiler), a `Maximum call stack size exceeded` error will be raised.

Reviewer: ```suggestion
A recursive function usually has a condition to stop calling itself and return a value, known as a _base case_.
If a base case is missing, in most cases, because it will call itself indefinitely, it would be able to run forever. 
In reality, in most of those situations, you'll end up with a "StackSize error": an error raised by the runtime because the _stack_ of function calls has grown beyond a predefined limit because each recursive call adds to this _stack_ until it returns (and it doesn't).
The message of this error is `Maximum call stack size exceeded`.
```

Reviewer: Counter examples are recursive functions that are killed from the outside, and tail-call optimized recursive functions (not possible in browsers and node at the moment). Async recursive functions also are options and technically don't require a base case.

Author: I'll take some time to improve on this, then.

Author: Or on retrospect, I'll just commit this for now. I'll add the tail-call optimization later.

```js
function recurse() {
  if (baseCondition) {
    // stop calling itself
    //...
  } else {
    recurse();
  }
}
```

> Recursive functions often can be used instead of `for` loops for more succinct code. For example, take a countdown.

Reviewer: ```suggestion
Recursive functions often can be used instead of `for` loops for more succinct code.
For example, take a countdown.
```
